fragments/add_dns_record.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# Update the DNS server with a record for this host
+
+set -eu
+set -x
+set -o pipefail
+
+DNS_UPDATE_KEY="%DNS_UPDATE_KEY%"
+
+if [ -z "$DNS_UPDATE_KEY" ]; then
+    echo "Skipping the DNS update because the key is empty."
+    exit
+fi
+
+if yum info python-dns; then
+    retry yum -y install python-dns
+else
+    retry yum -y install python2-dns
+fi
+
+HOSTNAME="$(hostname --fqdn)"
+
+for DNS_SERVER in %DNS_SERVERS%; do
```

Comment:
This should really only be a single nameserver.  All of the nameservers should agree and only one needs to be updated.  The others are slaves and will reflect the updates as they come.  See below.